cranelift-codegen/src/binemit/relaxation.rs

```diff
@@ -82,7 +85,7 @@ pub fn relax_branches(func: &mut Function, isa: &TargetIsa) -> CodegenResult<Cod
             // Record the offset for `ebb` and make sure we iterate until offsets are stable.
             if cur.func.offsets[ebb] != offset {
                 debug_assert!(
-                    cur.func.offsets[ebb] < offset,
+                    cur.func.offsets[ebb] < offset || could_have_shrunk,
                     "Code shrinking during relaxation"
                 );
```

Comment:
This assert seems pointless to me now. How could the `offset` ever be less if `could_have_shrunk` is false, since `offset` is calculated using the same values that are used to determine `could_have_shrunk`?

Reply:
Yeah, it's a fair point. @sunfishcode Should we just remove the assertion?

Reply:
Sure, removing it works for me.

Reply:
Cool, that even removes the need for the `could_have_shrunk` bool and in the signature of the relax function.